apps/condo/domains/user/schema/AuthenticateOrRegisterUserWithConfirmTokenService.js

```diff
@@ -0,0 +1,229 @@
+        mutation: 'authenticateOrRegisterUserWithConfirmToken',
+    },
+    WRONG_SENDER_FORMAT: {
+        ...COMMON_ERRORS.WRONG_SENDER_FORMAT,
+        mutation: 'authenticateOrRegisterUserWithConfirmToken',
+    },
+    DAILY_REQUEST_LIMIT_FOR_IP_REACHED: {
+        ...USER_ERRORS.DAILY_REQUEST_LIMIT_FOR_IP_REACHED,
+    },
+}
+
+const prepareCreateOrUpdateUserData = (user, userData, dvAndSender) => {
+    const payload = {}
+    for (const [key, value] of Object.entries(userData))  {
+        if (!value) continue
+        if (get(user, key)) continue
+        payload[key] = value
+    }
+    if (Object.keys(payload).length > 0) {
+        payload.dv = get(dvAndSender, 'dv')
+        payload.sender = get(dvAndSender, 'sender')
+    }
+    return payload
+}
+
+/**
+ * Checks that all required fields are present in a user data.
+ * @param {string[]} requiredFields
+ * @param {*} userData
+ * @return {{missingFields: string[], missingRequiredFields: boolean}}
+ */
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (isNil(value)) missingFields.push(field)
+        if (typeof value === 'string' && value.length < 1) missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const AuthenticateOrRegisterUserWithConfirmTokenService = new GQLCustomSchema('AuthenticateOrRegisterUserWithConfirmTokenService', {
+    types: [
+        {
+            access: true,
+            type: 'input AuthenticateOrRegisterUserWithConfirmTokenUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input AuthenticateOrRegisterUserWithConfirmTokenInput { dv: Int!, sender: SenderFieldInput!, confirmToken: ID!, userType: UserTypeType!, userData: AuthenticateOrRegisterUserWithConfirmTokenUserDataInput }',
+        },
+        {
+            access: true,
+            type: 'type AuthenticateOrRegisterUserWithConfirmTokenOutput { item: User, token: ID! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.authenticateOrRegisterUserWithConfirmToken,
+            schema: 'authenticateOrRegisterUserWithConfirmToken(data: AuthenticateOrRegisterUserWithConfirmTokenInput!): AuthenticateOrRegisterUserWithConfirmTokenOutput',
+            doc: {
+                summary: 'This mutation authorizes the user by token, after confirming the phone number.\n\n' +
+                    'If the user is not registered, then he will be created with the data that is passed in the payload (user data).\n\n' +
+                    'If the existing user is missing some fields, then these fields will be taken from the payload (user data) and updated.\n\n' +
```

Comment:
That's bad. What if I'll pass email here? You'll assign it to user without any questions?